knit/__init__.py

```diff
@@ -6,7 +6,7 @@
 from .env import CondaCreator, zip_path
 from .yarn_api import YARNAPI
 try:
-    from .dask_yarn import DaskYARNCluster
+    from dask_yarn import DaskYARNCluster
 except ImportError:    # pragma: no cover
     warnings.warn('dask/distributed not installed, '
                   'DaskYARNCluster not available')
```

Comment:
I know this is code from before, but I don't like this warning. This will show up for anyone that wants to use `knit` without dask. I'd rather silently not have `DaskYARNCluster` in the top level of knit.

Actually, at this point I'd deprecate `DaskYARNCluster` in the top-level namespace of knit completely. They are separate packages, and shouldn't be coupled in this way.

Reply:
Is there any way we can make this a slower deprecation? I know there aren't many users, but still it would seem normal to provide a warning before removing the name from knit.

Reply:
I meant deprecate it as we do other things - issue a warning for one release, and completely remove after that.

Reply:
The deprecation would show up every time - normally only done for a user-initiated action where they could have used an alternative method - plus dask_yarn will still not import without dask. Not sure how to make that pleasant.

Reply:
Subclass `DaskYARNCluster` in `knit`, and raise a warning in that classes's `__init__`?

Reply:
Yes, OK, that's not bad.